app/main/helpers/direct_award_helpers.py

```diff
@@ -29,3 +31,29 @@ def get_direct_award_projects(data_api_client, user_id, return_type="all", sort_
         if sort_by_key:
             res[return_type].sort(key=itemgetter(sort_by_key))
         return res[return_type]
+
+
+def set_direct_award_display_status(project, framework):
```

Comment:
I'm a little hesitant about burying the content in a helper function, as it makes it harder to find/change later if we want to tweak the text. However I understand the need to split this bit of complexity out of the main view.

I'd suggest grouping the content together at least, something like:

```
project_outcome_content = {
  "awarded": "Awarded",
  "cancelled": "The work has been cancelled",
  "none-suitable": "No suitable services found",
  "download_results": "Download results",
  "awaiting_outcome": "Tell us the results"
}
```

Then we can use that dict to do something like:

```
if project['outcome']:
  output = project_outcome_content[project['outcome']['result']]
elif not project['downloadedAt']:
  output = link_to(download_url, project_outcome_content['download_results'])
else:
  output = link_to(tell_us_the_outcome_url, project_outcome_content['awaiting_outcome'])
```

Whether that's simple enough to put back into the main view, I will leave up to you... 😸

Reply:
Agree with all of this - I should've taken a step back and thought about how I should structure things. Changing this has the added benefit of not requiring the `link_to` function from dmutils as well, so I'm going to close that.